### PR TITLE
fix: Fix duplicate breadcrumbs on Android

### LINKED ISSUE #1409: Duplicate Breadcrumbs
### Environment

How do you use Sentry?
sentry.io

Which SDK and version?
- react native 0.61.5
- @sentry/react-native: ^2.3.0

### Steps to Reproduce
```javascript
const dsn = Config.SENTRY_DSN;
const debug = false;
const release = `${DeviceInfo.getBundleId()}@${versionName}@${versionCode}+codepush:${codepushDist}`;
const dist = `${versionName}.${codepushDist}`;
const environment = isProductionEnv ? 'production' : 'staging';
const sampleRate = debug ? 1 : 0.5;
const tracesSampleRate = 0.2;
const maxBreadcrumbs = 150; 

    Sentry.init({
        dsn,
        debug,
        release,
        dist,
        environment,
        sampleRate,
        tracesSampleRate,
        maxBreadcrumbs,
    });
```

Set those initialization and surfing on my app normally until I get some errors.

### Expected Result

breadcrumb data is not logged twice.

### Actual Result
Looking at Breadcrumb section, I found that each breadcrumb data logged twice. `xhr` breadcrumb is logged by default by Sentry SDK. `btn_clicked` and `navigate` is manually logged, I mean, my code that logged it. Both of them logged twice. 

image

this is how I add a breadcrumb.
```
export function recordBreadcrumb(key, message, level = Sentry.Severity.Info) {
    Sentry.addBreadcrumb({
        category: key,
        message,
        level,
    });
}
...
function logButtonClicked(buttonOrFunctionName, value = {}){
   ...
   CrashLoggingUtils.recordBreadcrumb('btn_clicked', buttonOrFunctionName);
}
```

I've ensured that `logButtonClicked()` only called once a time.

### PR DESCRIPTION
**Pull request by jennmueng**
## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description

Clears breadcrumbs on the event before sending across the bridge on Android when `mechanism.handled` is true. See the comment for long explanation.

## :bulb: Motivation and Context

Fixes #1409 

## :green_heart: How did you test it?
Tested and reproduced via sample app the issue a couple times before the fix, and a couple times after to confirm that it no longer occurs. Some unit tests.

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: src/js/wrapper.ts [hunk 1]
```diff
@@ -40,16 +40,28 @@ export const NATIVE = {
...
 
+      /*
+        We do this to avoid duplicate breadcrumbs on Android as sentry-android applies the breadcrumbs
+        from the native scope onto every envelope sent through it. This scope will contain the breadcrumbs
+        sent through the scope sync feature. This causes duplicate breadcrumbs.
+        We then remove the breadcrumbs here but only when mechanism.handled is true. If it is handled == false,
+        this is a signal that the app would crash and android would lose the breadcrumbs by the time the app is restarted to read
+        the envelope.
+      */
+      if (event.exception?.values?.[0]?.mechanism?.handled) {
```
Comment (bruno-garcia) [inline src/js/wrapper.ts:54]: Do we need to check if scope sync is enabled? And the native layer is enabled etc? IIRC we have something like `enableNative` here

**Review by bruno-garcia — APPROVED** [review]
Just a note on the guard code but LGTM

File: src/js/wrapper.ts (hunk 1 above)
Comment (marandaneto) [reply src/js/wrapper.ts:54]: yep we agreed on that too

Comment (jennmueng) [reply src/js/wrapper.ts:54]: @bruno-garcia @marandaneto  `enableNative` is already checked beforehand and scope sync is always enabled, we haven't added an option to disable it yet.

File: src/js/wrapper.ts [hunk 2]
```diff
@@ -40,16 +40,28 @@ export const NATIVE = {
...
-      message: {
-        message: event.message,
-      },
-    };
-
     if (NATIVE.platform === "android") {
       const headerString = JSON.stringify(header);
 
+      /*
+        We do this to avoid duplicate breadcrumbs on Android as sentry-android applies the breadcrumbs
```
Comment (philipphofmann) [inline src/js/wrapper.ts:47]: Nice comment 🥇

**Review by marandaneto — APPROVED** [review]

**Pull request merged**